managers/views.py

```diff
@@ -160,3 +165,47 @@ def delete(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespons
         response = super().delete(request, args, kwargs)
         logger.info(f"Project with id: {kwargs['pk']} has been deleted by user with id: {self.request.user.pk}")
         return response
+
+
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(allowed_user_types=[CustomUser.UserType.ADMIN.name, CustomUser.UserType.MANAGER.name]),
+    name="dispatch",
+)
+class ManageTaskActivitiesInProjectView(FormView, UserIsManagerOfCurrentProjectMixin, SingleObjectMixin):
+    model = Project
+    template_name = "managers/project_task_activities.html"
+    form_class = TaskActivityForm
+
+    def _delete_task_activity_from_relation(self) -> None:
+        task_activity = get_object_or_404(TaskActivityType, pk=self.request.GET.get("task_activity"))
+        task_activity.projects.remove(self.object)
+
+    def _add_task_activities_to_relation(self, list_of_task_activities: list) -> None:
+        for task_activity_id in list_of_task_activities:
+            task_activity = get_object_or_404(TaskActivityType, pk=task_activity_id)
+            task_activity.projects.add(self.object)
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context = super().get_context_data()
+        context["project"] = self.object
+        context["activities_not_connected"] = TaskActivityType.objects.exclude(projects__in=self.kwargs["pk"])
+        context["task_activities"] = self.object.project_activities.all().order_by("name")
+        return context
+
+    def form_valid(self, form: TaskActivityForm) -> HttpRequest:
+        task_activity = form.save()
+        task_activity.projects.add(self.get_context_data()["project"])
+        return self.render_to_response(self.get_context_data(form=form))
+
+    def get(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        self.object = self.get_object()  # pylint: disable=attribute-defined-outside-init
```

Comment:
In `get` method `object` variable is not already initiated?

Reply:
Yes, FormView doesn't have self.object so I inherit from SingleObjectMixin and use all methods from there to work with object and queryset. What is more it give protection to this view only for manager of current project, because without SingleObjectMixin this mixin IsManagerOfCurrentProjectMixin doesn't work.

Reply:
I suggest trying to combine `FormView` and `DetailView`.

Reply:
As uppon I said that combine two generic view are very though or maybe even not possible. We should use mixins to get methods we want or make two views. 
https://docs.djangoproject.com/en/1.8/topics/class-based-views/mixins/#using-formmixin-with-detailview